install_config/aggregate_logging.adoc

```diff
@@ -3628,3 +3636,304 @@ cluster. Edit your non-cluster logging service (for example, `logging-es`,
 ----
 $ oc patch svc/logging-es -p '{"spec":{"selector":{"component":"es","provider":"openshift"}}}'
 ----
+
+[[optimal-state-cluster]]
+=== Cluster not in an optimal state
+
+The reasons that your cluster may not be considered to be in an optimal state are
+that the cluster health is not `green` and/or the cluster size specified in the
+cluster health does not match the number of pods running for that cluster (e.g.
+not all nodes have joined the cluster).
+
+To check which currently applies to your cluster:
+
+. Get the name of all pods for your particular cluster
++
+----
+$ oc get pods -l component=es[-ops] -o name
+----
+
+. Check the health of the cluster using one of the pods found above
+----
+$ oc exec -c elasticsearch <es_pod> -- es_cluster_health
+----
+
+The total number of nodes in the cluster displayed from the second command should
+match the number of pods returned in the first command.
+
+==== If the cluster is not `green`
+
+If the cluster is currently in a red state, see the section xref:recover-yellow-state[unable to recover to yellow].
+
+If the cluster is currently in a yellow state, see the section xref:recover-green-state[unable to recover to green].
+
+==== If the cluster size doesn't match pod count
+
+If the cluster size does not match the pod count, this likely means that the quorum
+for the cluster is not high enough and has resulted in another pod establishing
+a new cluster.
+
+For all the Deploymentconfigs, verify that the `NODE_QUORUM` environment variable
+is `(n / 2) + 1`. Where n is the number of nodes for the cluster.
+[NOTE]
+====
+For one node the quorum should be `1`, for two nodes it should be `2`, for three
+it should be `2`, etc.
+====
+
+Update any that do not match this formula and roll them out with:
+----
+$oc rollout latest dc/<dc_name>
+----
+
+[[node-unable-rollout]]
+=== Node unable to roll out
+
+To determine why a node had failed to roll out, check the events for that particular
+pod/deployment config (the follow commands assume you are currently in the project
```

Comment:
s/pod/deployment config (the follow commands assume you are currently in the project where the logging API objects exist):/pod and deployment config. The following commands assume you are currently in the project where the logging API objects exist: